lib/semian/adapter.rb

```diff
@@ -0,0 +1,34 @@
+module Semian
+  module Adapter
+    def semian_identifier
+      raise NotImplementedError.new("Semian adapters must implement a `semian_identifier` method")
+    end
+
+    private
+
+    def acquire_semian_resource(scope:)
+      return yield if @resource_acquired # Prevent a single adapter from acquiring the resource twice by nesting calls
```

Comment:
However what if you do something like:

```ruby
@redis1.pipelined do
  @redis1.set "omg", "hi"
  @redis1.set "lol", "hi"
  @redis2.set "omg", "sup"
end
```

Does `pipeline` actually end up acquiring a connection as well? I'm not completely familiar with how the client handles this situation.